.azure-pipelines/ci-build.yml

```diff
@@ -284,8 +339,103 @@ extends:
           #     Contents: 'scripts\**'
           #     TargetFolder: '$(Build.ArtifactStagingDirectory)'
  
-          - task: 1ES.PublishPipelineArtifact@1
-            displayName: 'Stage Artifacts: Nugets'
+          # - task: 1ES.PublishPipelineArtifact@1
+          #   displayName: 'Stage Artifacts: Nugets'
+          #   inputs:
+          #     artifactName: NuGets
+          #     targetPath: $(Build.ArtifactStagingDirectory)
+        - job: build_typescript
+          displayName: 'Build TypeScript Project'
+          pool:
+            name: Azure-Pipelines-1ESPT-ExDShared
+            image: ubuntu-latest
+            os: linux
+          steps:
+          # --- TypeScript Build, Test, and Package Steps ---
+          - task: NodeTool@0
+            displayName: 'Install Node.js'
+            inputs:
+              versionSpec: '20.x'
+          - script: npm ci
+            displayName: 'Install npm dependencies'
+            workingDirectory: '$(Build.SourcesDirectory)/typescript/'
+          - script: npm run build
+            displayName: 'Build project'
+            workingDirectory: '$(Build.SourcesDirectory)/typescript/'
+          - script: npm test
+            displayName: 'Run tests'
+            workingDirectory: '$(Build.SourcesDirectory)/typescript/' 
+          - script: npm pack --workspaces
+            displayName: 'Generate npm package'
+            workingDirectory: '$(Build.SourcesDirectory)/typescript/'
+          - task: CopyFiles@2
+            inputs:
+              sourceFolder: '$(Build.SourcesDirectory)/typescript/'
+              targetFolder: '$(Build.ArtifactStagingDirectory)/npm'
+              Contents: |
+                *.tgz
+            displayName: "Copy package.json and package gzipped tarballs"
+          templateContext:
+            outputs:
+            - output: pipelineArtifact
+              displayName: 'Move TypeScript package artifacts'
+              artifactName: npm
+              targetPath: $(Build.ArtifactStagingDirectory)/npm
+    - stage: deploy
+      condition: and(contains(variables['build.sourceBranch'], 'refs/tags/v'), succeeded())
+      dependsOn: [build] # We should only deploy from tagged builds.
+
+      jobs:
+        - job: publish_nuget_packages
+          displayName: 'Publish NuGet Packages'
+          pool:
+            name: Azure-Pipelines-1ESPT-ExDShared
+            image: windows-latest
+            os: windows
+          templateContext:
+              type: releaseJob
+              isProduction: true
+              inputs:
+                - input: pipelineArtifact
+                  artifactName: Nugets
+                  targetPath: $(Build.ArtifactStagingDirectory)/Nugets
+          steps:
```

Comment:
this needs to be under strategy.runOnce.deploy

Reply:
Resolved in #84